ansible_events_ui/api/activation.py

```diff
@@ -1,8 +1,40 @@
 """Activation API endpoints."""
-from fastapi import APIRouter
+import logging
+
+from fastapi import APIRouter, Depends
+from sqlalchemy import insert
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ansible_events_ui.db.dependency import get_db_session
+from ansible_events_ui.db.models import activations
+from ansible_events_ui.schemas import Activation
+
+logger = logging.getLogger("ansible_events_ui")
 
 __all__ = ("router",)
 
 router = APIRouter()
 
-# Implement activation API here.
+@router.post("/api/activation/")
+async def create_activation(
+    a: Activation,
+    db: AsyncSession = Depends(get_db_session),
+):
+    query = insert(activations).values(
+        name=a.name,
+        description=a.description,
+        rule_set_file_id=a.rule_set_file_id,
+        inventory_id=a.inventory_id,
+        execution_env_id=a.execution_env_id,
+        restart_policy_id=a.restart_policy_id,
+        playbook_id=a.playbook_id,
+        activation_enabled=a.activation_enabled,
+        extra_var_id=a.extra_var_id,
+        created=a.created,
```

Comment:
Question: Why are `created` and `last_modified` timestamps are passed from the client?

Reply:
Right, they should not. They should be autogenerated on the server side.